docs/HowTo/develop-fprime-libraries.md

```diff
@@ -0,0 +1,75 @@
+5. [Optional: F´ Module Directories](#optional-fmodule-directories)
+
+## F´ Library Structure
+
+In this section, you will learn about the expected structure of an F´ library. An F´ library must have a `library.cmake` and may have any of the following items:
+
+1. `cmake/toolchain` Folder and Toolchain Files
+2. `cmake/platform` Folder and Platform Files
+3. Module Directories and Modules (Components, Ports, Topologies, etc.)
+
+That means that a complete F´ library might look like the following:
+
+```bash
+library.cmake
+cmake/toolchain/my-toolchain.cmake
+cmake/platform/my-platform.cmake
+MyLibrary/Components/MyComponent/...
+MyLibrary/MyTopology/...
+```
+
+> Module directories shall be namespaced (i.e. `MyLibrary` above) to ensure that modules do not collide with F´ and other libraries. Placing directories like `Svc` directly at the root of the repository is *strongly* discouraged.
+
+You will see how to develop each of these parts of the library in the following sections.
+
+##  Required: `library.cmake`
+
+The `library.cmake` file is the only required file for an F´ library.  It consists of a series of `add_fprime_subdirectory` calls that include the various modules that are part of the library.  For libraries that just contain toolchains and platforms, this file may be empty. 
+
+**Sample `library.cmake`**
+```cmake
+####
+# library.cmake for MyLibrary
+####
+add_fprime_subdirectory("${CURRENT_CMAKE_LIST_DIR}/MyLibrary/Components/MyComponent")
+add_fprime_subdirectory("${CURRENT_CMAKE_LIST_DIR}/MyLibrary/MyTopology")
+```
+
+`library.cmake` should exist at the root of the library's directory structure.
+
+## Optional: Toolchain Folder and Toolchain Files
+
+F´ libraries often need to supply new toolchain integrations with F´. These libraries must define a `cmake/toolchain` folder and place toolchain files within that folder.  Any file ending in `.cmake` and placed within the `cmake/toolchain` folder will be available as a toolchain via `fprime-util generate <toolchain>`.
+
+The example toolchain above (`cmake/toolchain/ny-toolchain`) would be found as `my-toolchain` in `fprime-util`.
+
+```bash
+fprime-util generate my-toolchain
+```
+
+The `cmake/toolchain` folder may contain any number of toolchains and must be placed in the root of the library's directory structure.
+
+## Optional: Platform Folder and Platform Files
+
+In a similar manner to toolchains, platforms may be provided in the `cmake/platform` folder. Any file ending in `.cmake` and placed within the `cmake/platform` folder will be available to F´  to be loaded as a platform. Commonly, this means that a toolchain would include the line `set(FPRIME_PLATFORM <platform>)`.
+
+The example platform would thus need to include `set(FPRIME_PLATFORM "my-platform")`.
+
+The `cmake/platform` folder may contain any number of platform files and must be placed in the root of the library's directory structure.
+
+## Optional: F´ Module Directories
```

Comment:
This section should move higher in this doc, preferably near the structure section.

Reply:
Agreed